ui/apps/platform/src/Containers/DeferralConfiguration/VulnerabilitiesConfiguration.tsx

```diff
@@ -119,6 +139,61 @@ function getDefaultConfig(): VulnerabilitiesDeferralConfig {
     };
 }
 
+const validationSchema = yup.object({
+    expiryOptions: yup.object({
+        dayOptions: yup
+            .array()
+            .ensure()
+            .of(
+                yup.object({
+                    numDays: yup
+                        .number()
+                        .test(
+                            'isPositive',
+                            'Number of days must be greater than zero',
+                            (value) => typeof value === 'number' && value > 0
+                        )
+                        .required('Number of days must not be empty'),
+                    enabled: yup.boolean().required(),
+                })
+            )
+            .test('dayValuesAreUnique', (dayOptions, testContext) => {
+                if (!dayOptions) {
+                    return true;
+                }
+
+                const dayValueToIndexMap: Record<number, number> = {};
+                let error: yup.ValidationError | boolean = true;
```

Comment:
It looks like `error` is set to `true` as a default. But when we encounter a duplicate, the error changes its type. The initial setup was a tad tricky to follow. Maybe we could use `null` instead of a `boolean` for clarity? What do you think?

Reply:
Sorry I had a pending comment on this PR that never went through. The API for the yup validation functions is odd in that they return `true` if the validation succeeds, but an error otherwise.

It might be cleaner to not combine the types here though, and do an explicit `return true` at the end if `error` is `undefined`.